### PR TITLE
Feat: `lambda@edge` submodule

### PR DESCRIPTION
**Pull request by korenyoni**
## what
* Add submodule for creating Lambda@Edge functions
* Add test for Lambda@Edge submodule
* Ensure all tests run in parallel.
* Fix `versions.tf` in `examples/complete` according to TFLint

## why
* Lambda@Edge are a potentially common use case for this repo, and a submodule for creating Lambda@Edge functions for use with the root module should be created. 

## references
* N/A

## notes

Please note that a side effect of this change is that the test suite will now take ~30 minutes to run.

### CODE REVIEW

File: modules/lambda@edge/package.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+data "archive_file" "lambda_zip" {
+  for_each = local.functions
+
+  type             = "zip"
+  source_dir       = each.value.local_path
+  output_file_mode = "0666"
+  output_path      = "${path.module}/archives/${each.key}.zip"
```
Comment (osterman) [inline modules/lambda@edge/package.tf:7]: Can we also support inline files?

Comment (korenyoni) [reply modules/lambda@edge/package.tf:7]: Sure, i'll look into that. I don't think there's a reason we can't.

File: modules/lambda@edge/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+output "lambda_function_association" {
+  description = "TODO"
```
Comment (osterman) [inline modules/lambda@edge/outputs.tf:2]: FYI

File: modules/lambda@edge/package.tf (hunk 1 above)
Comment (korenyoni) [reply modules/lambda@edge/package.tf]: I've updated the schema.

File: modules/lambda@edge/outputs.tf (hunk 2 above)
Comment (korenyoni) [reply modules/lambda@edge/outputs.tf]: Created a description.

**Review by goruha — DISMISSED** [review]

**Review by goruha — APPROVED** [review]

**Review by goruha — DISMISSED** [review]

File: modules/lambda@edge/main.tf [hunk 3]
```diff
@@ -0,0 +1,94 @@
+locals {
+  enabled                   = module.this.enabled
+  destruction_delay_enabled = local.enabled && try(length(var.destruction_delay), 0) > 0
+  functions                 = local.enabled ? var.functions : {}
```
Comment (aknysh) [inline modules/lambda@edge/main.tf:4]: did you try to plan the module with `local.enabled=false`?
Terraform usually complains that both sides of the `:` operator must have the same types and the same item types.
If TF does that in this case, then the solution can be:

```
functions  = {for k,v in var.functions : k => v if local.enabled }
```

File: modules/lambda@edge/main.tf [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+module "role" {
+  for_each = local.functions
+
+  source  = "cloudposse/iam-role/aws"
+  version = "0.14.0"
+
+  use_fullname       = true
+  policy_description = "Allow ${module.function_label[each.key].id} Write CloudWatch Logs"
```
Comment (aknysh) [inline modules/lambda@edge/main.tf:54]:
```suggestion
  policy_description = "Allow ${module.function_label[each.key].id} write to CloudWatch Logs"
```

File: modules/lambda@edge/main.tf [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  role_description   = "IAM role for ${module.function_label[each.key].id}"
```
Comment (aknysh) [inline modules/lambda@edge/main.tf:55]:
```suggestion
  role_description   = "IAM role for ${module.function_label[each.key].id} Lambda function"
```

**Review by aknysh — COMMENTED** [review]
LGTM, a few nitpicks

File: modules/lambda@edge/main.tf (hunk 3 above)
Comment (korenyoni) [reply modules/lambda@edge/main.tf:4]: @aknysh yes, the test suite has a few cases where `enabled=false`, and there wasn't an issue.

**Review by aknysh — APPROVED** [review]
